### PR TITLE
simple idea to make rok8s-scripts compatible with docker older and ne…

### PR DESCRIPTION
**Pull request by ejether**
…wer docker versions

Its a bit of a monkey patch, but it might be just the trick to move past this hurdle?

### CODE REVIEW

**Comment by ejether** [comment]
Good point. I made the assumption that on Circle all 1.0 builds would be on Docker 1.9 but didn’t think about other cases outside of Circle.

**Comment by philipcristiano** [comment]
Eventually would be nice to have a `docker-version-check` script called at the beginning of docker scripts that would set feature flags. In this case something like

```
docker < N:
    DOCKER_FLAG_FORCE_TAG="-f"
```

This gets a bit messy though to keep doing in shell. [spotify/docker-client](https://github.com/spotify/docker-client) is a thing (although java) so we might be able to find a compatibility library somewhere.

File: bin/docker-push [hunk 1]
```diff
@@ -31,7 +31,11 @@ then
   exit 1
 fi
 
-docker tag -f "${DOCKERTAG}:latest" "${EXTERNAL_REGISTRY_BASE_DOMAIN}/${REPOSITORY_NAME}:${CI_REF}"
+DOCKER_TAG_COM="docker tag"
+docker version | grep -e Version | head -n 1 | grep  1.9 && DOCKER_TAG_COM="${DOCKER_TAG_COM} -f "
```
Comment (mecampbellsoup) [inline bin/docker-push]: So `DOCKER_TAG_COM` only gets converted from `"docker tag"` to `"docker tag -f"` if the `grep 1.9` succeeds, right? (Meaning the docker version *matches* `1.9` anywhere in that version string...?)

Are you looking specifically for version like 17.01.9 and below? I'm running 17.06.1-ce-rc1 currently on my local machine. I guess I'm asking about the `grep 1.9` command specifically and where "1.9" came from.

Comment (philipcristiano) [inline bin/docker-push]: Good catch for the versions, with the change for CE/EE, 1.9 likely isn't safe enough to use.

Comment (ejether) [inline bin/docker-push]: I'm amending this to be safer. Update coming shortly.

File: bin/docker-push [hunk 2]
```diff
@@ -33,10 +33,10 @@ fi
 
 DOCKER_TAG_COM="docker tag"
 version=$(docker version | grep -e Version | head -n 1 | awk '{ print $2 }')
-major=$(echo ${version} | awk -F. '{ print $1 }')
-minor=$(echo ${version} | awk -F. '{ print $2 }')
+major=$(echo "${version}" | awk -F. '{ print $1 }')
+minor=$(echo "${version}" | awk -F. '{ print $2 }')
 
-if [[ ${minor} -le 9 && ${major} -eq 1 ]] || [ ${major} -eq 0 ]
+if [[ "${minor}" -le 9 && "${major}" -eq 1 ]] || [ "${major}" -eq 0 ]
```
Comment (mecampbellsoup) [inline bin/docker-push]: How does e.g. `"08"` minor version evaluate in `"${minor}" -le 9`?

**Comment by ejether** [comment]
Poorly. Good catch.

File: bin/docker-push (hunk 2 above)
Comment (ejether) [inline bin/docker-push]: Poorly. Good catch.

**Pull request merged**